gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -1384,17 +1379,37 @@ public void proguard()
         loggingManager.captureStandardError(LogLevel.WARN);
 
         // Run ProGuard with the collected configuration.
-        new ProGuard(getConfiguration()).execute();
+        new ProGuard(getConfiguration(getConfigurationFiles().getElements().get())).execute();
+
+    }
 
+    private Provider<Configuration> getConfigurationProvider() {
```

Comment:
ProGuard Java code style normally has `{` on a new line. Could you update them in your newly added code?

Reply:
Done in f70c204